src/main/java/bio/terra/service/filedata/flight/delete/DeleteFileDirectoryStep.java

```diff
@@ -25,11 +27,15 @@ public DeleteFileDirectoryStep(FireStoreDao fileDao,
 
     @Override
     public StepResult doStep(FlightContext context) {
-        boolean found = fileDao.deleteDirectoryEntry(dataset, fileId);
-        DeleteResponseModel.ObjectStateEnum stateEnum =
-            (found) ? DeleteResponseModel.ObjectStateEnum.DELETED : DeleteResponseModel.ObjectStateEnum.NOT_FOUND;
-        DeleteResponseModel deleteResponseModel = new DeleteResponseModel().objectState(stateEnum);
-        FlightUtils.setResponse(context, deleteResponseModel, HttpStatus.OK);
+        try {
+            boolean found = fileDao.deleteDirectoryEntry(dataset, fileId);
+            DeleteResponseModel.ObjectStateEnum stateEnum =
+                (found) ? DeleteResponseModel.ObjectStateEnum.DELETED : DeleteResponseModel.ObjectStateEnum.NOT_FOUND;
+            DeleteResponseModel deleteResponseModel = new DeleteResponseModel().objectState(stateEnum);
+            FlightUtils.setResponse(context, deleteResponseModel, HttpStatus.OK);
+        } catch (FileSystemRetryException rex) {
```

Comment:
I'm an outsider observer of this code, so I may be mistaken, but I thought that `File.stepWithRetry()` implemented logic around calls to `doStep()` to avoid adding try/catch to steps that convert exceptions into results. Or is there a reason why this conversion wouldn't be generally applicable?

Reply:
I debated this. I could have done as you suggest and derived FileSystemRetryException from the Stairway RetryException. Then the Stairway could would automatically handle the catch and retry. It is less code and maybe cleaner.

I chose the explicit try-catch for readability. It lets you look at the step and understand that retries are being done under a specific condition.

I could go either way on it. Other reviewers, please weigh in on the question.